### PR TITLE
React rework

### PR DESCRIPTION
**Pull request by railsstudent**

### CODE REVIEW

**Comment by tonyhhyip** [comment]
Please upload some screen capture if possible

**Comment by railsstudent** [comment]
Fixing it based on feedback

**Comment by tonyhhyip** [comment]
@railsstudent Do you forget to git push?

**Comment by railsstudent** [comment]
screen shot 2017-01-24 at 6 18 36 am
screen shot 2017-01-24 at 11 55 06 pm

**Comment by railsstudent** [comment]
screen shot 2017-01-24 at 11 54 57 pm

File: .yarnclean [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (tonyhhyip) [inline .yarnclean]: Please remove this file with `git rm --cached .yarnclean`

File: assets/js/InputCharacterComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      textFieldValue: newValue,
+      inputText: newValue.split('\n')
+    });
+  }
+
+  render() {
+    let numRows = 8;
+    return(
+      <div style={styles.container}>
+        <div style={styles.bigfontStyle}>
```
Comment (tonyhhyip) [inline assets/js/InputCharacterComponent.jsx]: Font family of freehkfonts is missed

File: package.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    "webpack-merge": "2.3.1"
+  },
+  "dependencies": {
+    "material-ui": "0.16.7",
+    "react": "15.4.2",
+    "react-addons-css-transition-group": "15.4.2",
+    "react-dom": "15.4.2",
+    "react-tap-event-plugin": "2.0.1"
+  },
+  "babel": {
```
Comment (tonyhhyip) [inline package.json]: You may remove the part as you have the .babelrc

File: public/index.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+      })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
+
+      ga('create', 'UA-47867649-10', 'auto');
+      ga('send', 'pageview');
+
+    </script>
+
+    <!--script src="/js/app.js" type="text/javascript"></script -->
```
Comment (tonyhhyip) [inline public/index.html]: Please remove commented lines

File: assets/js/CreditComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  },
+  licenseStyle: {
+      fontSize: '24px',
+      marginLeft: '10px'
+  }
+};
+
+class CreditComponent extends BaseMUI {
+
+  constructor(props) {
```
Comment (tonyhhyip) [inline assets/js/CreditComponent.jsx]: Remove this do nothing constructor

**Review by tonyhhyip — CHANGES_REQUESTED** [review]
Great Job, just some minor issue and some way to be improved. Please take a look at the lines I comment on and consider to fix it

File: .yarnclean (hunk 1 above)
Comment (railsstudent) [reply .yarnclean]: Done

File: assets/js/InputCharacterComponent.jsx (hunk 2 above)
Comment (railsstudent) [reply assets/js/InputCharacterComponent.jsx]: Done

File: package.json (hunk 3 above)
Comment (railsstudent) [reply package.json]: Done

File: public/index.html (hunk 4 above)
Comment (railsstudent) [reply public/index.html]: Done

File: assets/js/CreditComponent.jsx (hunk 5 above)
Comment (railsstudent) [reply assets/js/CreditComponent.jsx]: Done

**Comment by railsstudent** [comment]
Please review the changes again.

**Review by tonyhhyip — APPROVED** [review]

**Comment by tonyhhyip** [comment]
Some place can improve like code style, I will handle it

**Comment by railsstudent** [comment]
Sure, thanks.

**Pull request merged**